README.md

```diff
@@ -6,34 +6,81 @@
+
+Here is a very simple example DAG:
 
 ```julia
 using Dagger
 
-p = delayed(f; options...)(42)
-q = delayed(g)(p)
-r = delayed(h)(53)
+add1(value) = value+1
+add2(value) = value+2
+combine(a...) = sum(a)
+
+p = delayed(add1)(4)
+q = delayed(add2)(p)
+r = delayed(add1)(3)
 s = delayed(combine)(p, q, r)
+
+collect(s)
 ```
 The connections between nodes `p`, `q`, `r` and `s` is represented by this dependency graph:
 
 ![graph](https://user-images.githubusercontent.com/25916/26920104-7b9b5fa4-4c55-11e7-97fb-fe5b9e73cae6.png)
 
-`delayed(f; options...)`
 
-Returns a function which when called creates a `Thunk` object representing a call to function `f` with the given arguments. If it is called with other thunks as input, then they form a graph with input nodes directed at the output. The function `f` get the result of evaluating the input thunks.
+The final result is the obvious consequence of the operation
+
+ `add1(4)` + `add2(add1(4))` + `add1(3)`
+
+ `(4 + 1)` + `((4 + 1) + 2)` + `(3 + 1)` = 16
 
 To compute and fetch the result of a thunk (say `s`), you can call `collect(s)`. `collect` will fetch the result of the computation to the master process. Alternatively, if you want to compute but not fetch the result you can call `compute` on the thunk. This will return a `Chunk` object which references the result. If you pass in a `Chunk` objects as an input to a delayed function, then the function will get executed with the value of the `Chunk` -- this evaluation will likely happen where the input chunks are, to reduce communication.
+
+The key point is that, for each argument to a node, if the argument is a `Thunk`, it'll be executed before this node and its result will be passed into the function `f` provided.
+If the argument is *not* a `Thunk` (just some regular Julia object), it'll be passed as-is to the function `f`
+
+
  
 Options to `delayed` are:
 - `get_result::Bool` -- return the actual result to the scheduler instead of `Chunk` objects. Used when `f` explicitly constructs a Chunk or when return value is small (e.g. in case of reduce)
 - `meta::Bool` -- pass the input “Chunk” objects themselves to `f` and not the value contained in them - this is always run on the master process
 - `persist::Bool` -- the result of this Thunk should not be released after it becomes unused in the DAG
 - `cache::Bool` -- cache the result of this Thunk such that if the thunk is evaluated again, one can just reuse the cached value. If it’s been removed from cache, recompute the value.
 
+
+### Polytree
+
+[Polytrees](https://en.wikipedia.org/wiki/Polytree "Polytrees") are easily supported by Dagger. To do the trick, pass all the head nodes `Thunks` in a `Thunk` that you will create as a top node for the graph end compute it.
+```
+group(x...) = [x...]
+top_node = delayed(group)(head_nodes...)
+compute(top_node)
+```
+
+# DAG creation interface
```

Comment:
I think this could stand to move back up to before the examples, since the first section after this isn't about DAG creation but about scheduling.

Reply:
This is still down here in the diff?